scripts/build_desktop_app_with_firebase.py

```diff
@@ -0,0 +1,281 @@
+          If its not specified, cmake's default is used (most likely Debug).
+   target_format (str): If specified, build for this targetformat ('frameworks' or 'libraries').
+  """
+  # Cmake configure
+  cmd = ['cmake', '-S', '.', '-B', build_dir]
+  cmd.append('-DFIREBASE_CPP_SDK_DIR={0}'.format(sdk_source_dir))
+
+  # If generator is not specifed, default for platform is used by cmake, else
+  # use the specified value
+  if config:
+    cmd.append('-DCMAKE_BUILD_TYPE={0}'.format(config))
+    # workaround, absl doesn't build without tests enabled
+    cmd.append('-DBUILD_TESTING=off')
+
+  if platform.system() == 'Linux' and arch == 'x86':
+    # Use a separate cmake toolchain for cross compiling linux x86 builds
+    vcpkg_toolchain_file_path = os.path.join(sdk_source_dir, 'external', 'vcpkg',
+                                             'scripts', 'buildsystems', 'linux_32.cmake')
+  else:
+    vcpkg_toolchain_file_path = os.path.join(sdk_source_dir, 'external',
+                                           'vcpkg', 'scripts',
+                                           'buildsystems', 'vcpkg.cmake')
+
+  cmd.append('-DCMAKE_TOOLCHAIN_FILE={0}'.format(vcpkg_toolchain_file_path))
+
+  vcpkg_triplet = get_vcpkg_triplet(arch, msvc_runtime_library)
+  cmd.append('-DVCPKG_TARGET_TRIPLET={0}'.format(vcpkg_triplet))
+
+  if platform.system() == 'Windows':
+    # If building for x86, we should supply -A Win32 to cmake configure
+    # Its a good habit to specify for x64 too as the default might be different
+    # on different windows machines.
+    cmd.append('-A')
+    cmd.append('Win32') if arch == 'x86' else cmd.append('x64')
+
+    # Use our special cmake option for /MD (dynamic).
+    # If this option is not specified, the default value is /MT (static).
+    if msvc_runtime_library == "static":
+      cmd.append('-DMSVC_RUNTIME_LIBRARY_STATIC=ON')
+
+  if (target_format):
+    cmd.append('-DFIREBASE_XCODE_TARGET_FORMAT={0}'.format(target_format))
+  print("Running {0}".format(' '.join(cmd)))
+  subprocess.run(cmd, cwd=app_dir, check=True)
+
+  #CMake build
+  cmd = ['cmake', '--build', build_dir, '-j', str(os.cpu_count()), '--config', config]
+  print("Running {0}".format(' '.join(cmd)))
+  subprocess.run(cmd, cwd=app_dir, check=True)
+
+def build_app_with_prebuilt(app_dir, sdk_prebuilt_dir, build_dir, arch,
+                            msvc_runtime_library='static', config=None):
+  """Build desktop app directly against the prebuilt firebase C++ libraries.
+
+  Since this invovles a cmake configure, it is advised to run this on a clean
+  build directory.
+
+  Args:
+   app_dir (str): Path to directory containing application's CMakeLists.txt.
+   sdk_prebuilt_dir (str): Path to prebuilt firebase C++ libraries.
+   build_dir (str): Output build directory.
+   arch (str): Platform Architecture (example: 'x64').
+   msvc_runtime_library (str): Runtime library for MSVC (eg: 'static', 'dynamic').
+   config (str): Release/Debug config.
```

Comment:
Might be worth specifying examples here.  Like this should be either the string "Release" or "Debug".

Reply:
Done